api/incubator/src/main/java/io/opentelemetry/api/incubator/config/ConfigProvider.java

```diff
@@ -30,6 +32,22 @@ public interface ConfigProvider {
   @Nullable
   DeclarativeConfigProperties getInstrumentationConfig();
 
+  /**
+   * Returns the {@link DeclarativeConfigProperties} for a specific instrumentation by name. If no
+   * configuration is available for the given name, an empty {@link DeclarativeConfigProperties} is
+   * returned.
+   *
+   * @param name the name of the instrumentation
+   * @return the {@link DeclarativeConfigProperties} for the given instrumentation name
+   */
+  default DeclarativeConfigProperties get(String name) {
```

Comment:
I'm a bit worried that if `ConfigProvider` is expanded in the future to have other functions, it won't be clear that `get` is sugar for `getInstrumentationConfig`.

Can remove this risk with naming `getJavaInstrumentationConfig(String name)`, which I think is more clear that this method is about accessing a particular java instrumentation library's config.

Looking at this, I wonder if we should scrap #7920 in favor of an approach like this PR @trask.

This approach is a bit of a "have your cake and eat it to", since it still allows users to determine if `.instrumentation/development` was set, but provides syntactic sugar for the common case where you don't actually care.